### PR TITLE
RFC: Better Poro Resource Test

### PR DESCRIPTION
**Pull request by bf4**
Per https://github.com/cerebris/jsonapi-resources-site/issues/45#issuecomment-558638279
I'm trying to flesh out plain old resource support in JR 0.10.x

Here for comment and if good I'll bring up to standards to be mergeable

 ### All Submissions:

- [x] I've checked to ensure there aren't other open [Pull Requests](https://github.com/cerebris/jsonapi-resources/pulls) for the same update/change.
- [ ] I've submitted a [ticket](https://github.com/cerebris/jsonapi-resources/issues) for my issue if one did not already exist.
- [ ] My submission passes all tests. (Please run the full test suite locally to cut down on noise from travis failures.)
- [ ] I've used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message or the description.
- [ ] I've added/updated tests for this change.

 ### New Feature Submissions:

- [ ] I've submitted an issue that describes this feature, and received the go ahead from the maintainers.
- [ ] My submission includes new tests.
- [ ] My submission maintains compliance with [JSON:API](http://jsonapi.org/).

 ### Bug fixes and Changes to Core Features:

- [ ] I've included an explanation of what the changes do and why I'd like you to include them.
- [ ] I've provided test(s) that fails without the change.

 ### Test Plan:

 ### Reviewer Checklist:
- [ ] Maintains compliance with JSON:API
- [ ] Adequate test coverage exists to prevent regressions

### CODE REVIEW

File: test/fixtures/active_record.rb [hunk 1]
```diff
@@ -1598,14 +1679,6 @@ def find_breeds_by_keys(keys, options = {})
...
-
-class BreedResource < PoroResource
-
-  attribute :name, format: :title
-
-  # This is unneeded, just here for testing
-  routing_options param: :id
 
   def _save
     super
```
Comment (bf4) [inline test/fixtures/active_record.rb:1684]: hmm

**Comment by bf4** [comment]
cc @scottgonzalez here's my thinking (btw, test failures seem unrelated)

1. JR doesn't need to support a poro resource, but can document and test one and point people to use that if they wish
2. this refactor at time of comment begins to support a pattern for a poro resource or general subclassing of basic resource
3. next step would either be not to rely on globals in the poro resource to mimic persistence or have two basic resource subclasses 1) non ar persistence 2) non-persisted. The non-persisted one could use technique @scottgonzalez mentioned in https://github.com/cerebris/jsonapi-resources-site/issues/45#issuecomment-558638279

  > If the resources are not findable (perhaps they do not actually persist), you can store the created model in context in an `after_create` callback and then return that model in the finder (you can access `context` via the `options` argument).

**Review by lgebhardt — APPROVED** [review]
LGTM

The travis failures came in with the introduction of the postgres testing.

**Comment by lgebhardt** [comment]
@bf4 What do you wish to add to make this mergeable?

**Comment by bf4** [comment]
@lgebhardt I'm not sure what I might have had in mind when I first wrote the PR.   Maybe add more tests around when the various 'NotImplemented' methods are used?  I'd like to remove the globals, too. But really, I'm just trying to share what I've learned in trying to use 0.10.x

**Comment by lgebhardt** [comment]
@bf4 Sounds good, and thanks for the PR. I merged this as is and we can update things later.

**Comment by bf4** [comment]
xref https://github.com/cerebris/jsonapi-resources/issues/1306

**Pull request merged**